core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala

```diff
@@ -187,4 +188,64 @@ class TopicCommandTest extends ZooKeeperTestHarness with Logging with RackAwareT
     }
     checkReplicaDistribution(assignment, rackInfo, rackInfo.size, alteredNumPartitions, replicationFactor)
   }
+
+  @Test
+  def testDescribeTopicsMarkedForDeletion() {
+    val brokers = List(0)
+    val topic = "testtopic"
+    val markedForDeletion = "MarkedForDeletion"
+    TestUtils.createBrokersInZk(zkUtils, brokers)
+
+    val createOpts = new TopicCommandOptions(Array("--partitions", "1",
+      "--replication-factor", "1",
+      "--topic", topic))
+    TopicCommand.createTopic(zkUtils, createOpts)
+
+    // delete the broker first, so when we attempt to delete the topic it gets into
+    // "marked for deletion"
+    TestUtils.deleteBrokersInZk(zkUtils, brokers)
+    val deleteOpts = new TopicCommandOptions(Array("--topic", topic))
+    TopicCommand.deleteTopic(zkUtils, deleteOpts)
+
+    def unitDescribeConfig() {
+      val describeOpts = new TopicCommandOptions(Array("--describe"))
+      TopicCommand.describeTopic(zkUtils, describeOpts)
+    }
+    val outputConfig = TestUtils.grabConsoleOutput(unitDescribeConfig)
+    assertTrue(outputConfig.contains(topic) && outputConfig.contains(markedForDeletion))
+
+    def unitDescribeNoConfig() {
+      val describeOpts = new TopicCommandOptions(Array("--describe", "--unavailable-partitions"))
+      TopicCommand.describeTopic(zkUtils, describeOpts)
+    }
+    val outputNoConfig = TestUtils.grabConsoleOutput(unitDescribeNoConfig)
+    assertTrue(outputNoConfig.contains(topic) && outputNoConfig.contains(markedForDeletion))
+  }
+
+  @Test
+  def testListTopicsMarkedForDeletion() {
```

Comment:
I suggest merging this test with the other one. Even though it's a good practice to write separate tests for separate functionality, starting ZK and a broker twice for what we're testing here is not worth it.